api/renterd/api.go

```diff
@@ -49,6 +57,10 @@ type server struct {
 	s  Syncer
 	cm ChainManager
 	tp TransactionPool
+
+	mu        sync.Mutex
+	contracts []rhp.Contract
+	hostSets  map[string][]types.PublicKey
```

Comment:
for now, we can just return the contract in the API response and not save anything server-side. We can skip the host set stuff too; I'm not sure it's the best design